src/viking.js

```diff
@@ -1,11 +1,99 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+
+    attack() {
+        return this.strength;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name;
+    }
+
+    attack() {
+        return this.strength;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health <= 0) {
+            return `${this.name} has died in act of combat`;
+        } else {
+            return `${this.name} has received ${damage} points of damage`;
+        }
+    }
+    battleCry() {
+        return "Odin Owns You All!";
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health <=0){
+            return "A Saxon has died in combat";
+        } else {
+            return `A Saxon has received ${damage} points of damage`;
+        }   
+    }
+}
 
 // War
-class War {}
+class War {
+    constructor() {
+        this.vikingArmy = [];
+        this.saxonArmy = [];
+    }
+
+    addViking(viking) {
+            this.vikingArmy.push(viking);
+    }
+
+    addSaxon(saxon) {
+        this.saxonArmy.push(saxon);
+    }
+    vikingAttack() {
+
+        const randomVikingIndex = Math.floor(Math.random() * this.vikingArmy.length);
+        const randomSaxonIndex = Math.floor(Math.random() * this.saxonArmy.length);
+        const randomSaxon =  this.saxonArmy[randomSaxonIndex];
+        const result = randomSaxon.receiveDamage(this.vikingArmy[randomVikingIndex].strength);
```

Comment:
Esto esta bien! aunque es un poco lioso!
Te lo pongo como lo haria yo:

```suggestion
        const randomViking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
        const randomSaxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
        const result = randomSaxon.receiveDamage(randomViking.strength);
```

Asi te queda mas uniforme la informacion, pero lo que has hecho antes esta perfect!